lib/ansible/plugins/inventory/yc_compute.py

```diff
@@ -0,0 +1,304 @@
+# Copyright (c) 2019 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+    name: yc_compute
+    plugin_type: inventory
+    short_description: Yandex.Cloud Compute inventory source
+    requirements:
+      - yandexcloud==0.10.1
+    extends_documentation_fragment:
+      - constructed
+      - inventory_cache
+    description:
+      - Pull inventory from Yandex Cloud Compute.
+      - Uses a YAML configuration file that ends with yc_compute.(yml|yaml) or yc.(yml|yaml).
+    version_added: "2.10"
+    options:
+      auth_kind:
+        description: The type of credential used.
+        required: True
+        choices: ['oauth', 'serviceaccountfile']
+        env:
+          - name: YC_ANSIBLE_AUTH_KIND
+      oauth_token:
+        description: OAUTH token string. See U(https://cloud.yandex.com/docs/iam/concepts/authorization/oauth-token).
+        type: string
+        no_log: True
+        env:
+          - name: YC_ANSIBLE_OAUTH_TOKEN
+      service_account_file:
+        description:
+          - The path of a Service Account JSON file. Must be set if auth_kind is "serviceaccountfile".
+          - "Service Account JSON file can be created by C(yc) tool:"
+          - C(yc iam key create --service-account-name my_service_account --output my_service_account.json)
+        type: path
+        env:
+          - name: YC_ANSIBLE_SERVICE_ACCOUNT_FILE
+      service_account_contents:
+        description: Similar to service_account_file. Should contain raw contents of the Service Account JSON file.
+        type: string
+        env:
+          - name: YC_ANSIBLE_SERVICE_ACCOUNT_CONTENTS
+        no_log: True
+      hostnames:
+        description:
+          - The list of methods for determining the hostname.
+          - Several methods can be tried one by one. Until successful hostname detection.
+          - Currently supported methods are 'public_ip', 'private_ip' and 'fqdn'.
+          - Any other value is parsed as a jinja2 expression.
+        default: ['public_ip', 'private_ip', 'fqdn']
+        type: list
+      folders:
+        description: List of Yandex.Cloud folder ID's to list instances from.
+        type: list
+        required: True
```

Comment:
This is documented as `required: True`, but below `self.get_option('folders') or []` happens, making it seem optional. Is that right?

Reply:
Folders section is required but still can be empty. In this case get_option will return None. Maybe I should raise exception in that case, but for now plugin just returns empty results.